db.go

```diff
@@ -103,6 +103,9 @@ type conn struct {
 	drv       *txDriver
 	saves     uint
 	savePoint SavePoint
+
+	cancel func()
+	ctx    interface{ Done() <-chan struct{} }
```

Comment:
inlining this interface to avoid calling into `context.Context`, which would break backwards compatability with versions go that predate the context pkg